src/gui/traymenu.cpp

```diff
@@ -169,9 +177,20 @@ void TrayMenu::clearAllActions()
 
 void TrayMenu::setViModeEnabled(bool enabled)
 {
+    if (enabled) {
+        m_emacsMode = false;
+    }
     m_viMode = enabled;
 }
 
+void TrayMenu::setEmacsModeEnabled(bool enabled)
+{
+    if (enabled) {
+        m_viMode = false;
+    }
+    m_emacsMode = enabled;
```

Comment:
Something to consider is using an enum to store the current navigation style. Also might be good to change the two check boxes in preferences into a combo-box or radio button group with three options.